File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface BusinessEntityForm {
<<<<<<< HEAD
=======
    PersonType: StringEditor;
>>>>>>> revert
    Title: StringEditor;
    FirstName: StringEditor;
    MiddleName: StringEditor;
    LastName: StringEditor;
<<<<<<< HEAD
    FullName: StringEditor;
    Suffix: StringEditor;
=======
    Suffix: StringEditor;
    AddressType: StringEditor;
>>>>>>> revert
}

export class BusinessEntityForm extends PrefixedContext {
    static formKey = 'Company.BusinessEntity';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BusinessEntityForm.init)  {
            BusinessEntityForm.init = true;

            var w0 = StringEditor;

            initFormType(BusinessEntityForm, [
<<<<<<< HEAD
=======
                'PersonType', w0,
>>>>>>> revert
                'Title', w0,
                'FirstName', w0,
                'MiddleName', w0,
                'LastName', w0,
<<<<<<< HEAD
                'FullName', w0,
                'Suffix', w0
=======
                'Suffix', w0,
                'AddressType', w0
>>>>>>> revert
            ]);
        }
    }
}

File: Ser2023/Company/Company.Web/Modules/Company/BusinessEntity/BusinessEntityForm.cs
using Serenity.ComponentModel;
using System.Collections.Generic;
using System.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.BusinessEntity")]
[BasedOnRow(typeof(BusinessEntityRow), CheckNames = true)]
public class BusinessEntityForm
{
    [Category("Person")]
    public string PersonType { get; set; }
    public string Title { get; set; }
    public string FirstName { get; set; }
    public string MiddleName { get; set; }
    public string LastName { get; set; }
    public string Suffix { get; set; }
    //public string AddressTypeName { get; set; }

    [Category("Address Type")]
    //[AddressTypeEditor]
    public List<AddressRow> AddressType { get; set; }  

    //[Category("Addresses")]
    //[BusinessEntityAddressEditor]
    //public List<BusinessEntityAddressRow> BusinessEntityAddresses { get; set; }
}